src/QsCompiler/LanguageServer/Program.cs

```diff
@@ -181,7 +181,7 @@ internal static QsLanguageServer ConnectViaStdInOut(string? logFile = null)
             return new QsLanguageServer(Console.OpenStandardOutput(), Console.OpenStandardInput());
         }
 
-        internal static QsLanguageServer ConnectViaNamedPipe(string writerName, string readerName, string? logFile = null)
+        internal static QsLanguageServer ConnectViaNamedPipe(string writerName, string readerName, string? logFile = null, string? notebookReferencesDir = null)
```

Comment:
Having to pass this all the way through everything is ugly, but I'm not sure what else to do. It's tempting to just have a hacky command (like the get diagnostics command) to set this, but that's also not very pretty

Reply:
Looks like #1491 runs the server in a new process, so we might need to do something else there, too (maybe a command line argument). Right now I can't think of a better solution.

Reply:
That's a good point, thanks. What do you think about a `workspace/executeCommand` command for setting this? (Similar to the "get diagnostics" one, for example.) That would make this explicitly for tests/debugging and not part of the public api

Reply:
I think to me it feels like something that only the program launching the LSP server should be able to set, rather than the LSP client. For example if the server is running on a different host than the client, it's weird for the client to be able to tell the server where to look for binaries locally.

How about an environment variable? It seems like it's kind of analagous to looking up something in `$PATH` or `$LD_LIBRARY_PATH`.

Reply:
The analogy to `$LD_LIBRARY_PATH` is pretty convincing, thanks. I'll try that

Reply:
Pushed up a little commit (e7ba09da69b6), is that what you were thinking?

Reply:
I was thinking that the environment variable would replace the QsLanguageServer constructor parameter and all of the fields storing the path; in NotebookProjectLoader we would read the environment variable directly or fall back to the assembly path. I think that minimizes the number of places we need to pass the path around.